### DISCUSSION TITLE
Solution for authentication that works from the Tesla browser.

### DISCUSSION THREAD

**Discussion by BowlesCR** [teslamate-org/teslamate#1519] [Ideas]
Related to #413 and #905, I found a quick and dirty way to do auth that will work from the Tesla browser.
The https://user:pass@host/ method seems to no longer work, and had some major drawbacks anyway.

By (ab)using the Grafana API's "login ping" endpoint, we can secure the teslamate path(s) behind Grafana's form-based authentication.

Snippet from my nginx conf:

    location /grafana/ {
        proxy_cookie_path /grafana /;
        proxy_pass http://127.0.0.1:3000;
    }

    location /live/websocket {
        proxy_pass http://127.0.0.1:4000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        auth_request /grafana/api/login/ping;
    }

    location / {
        proxy_pass http://127.0.0.1:4000;
        auth_request /grafana/api/login/ping;
        error_page 401 = @grafanalogin;
    }

    location @grafanalogin {
        return 302 /grafana/login/;
    }

**Comment by adriankumpf** [comment]
Interesting! I'll move this to [💡Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas)
